File: app/src/main/java/ir/ghararemaghzha/game/dialogs/RulesDialog.java
package ir.ghararemaghzha.game.dialogs;

import android.app.Dialog;
import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;

import com.google.android.material.textview.MaterialTextView;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

import ir.ghararemaghzha.game.R;

public class RulesDialog extends Dialog {

    private MaterialTextView text;
    private Context context;


    public RulesDialog(@NonNull Context context) {
        super(context);
        this.context=context;
    }


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.dialog_rules);
        text = findViewById(R.id.rules_text);
        text.setText(getText());
        findViewById(R.id.rules_close).setOnClickListener(v->dismiss());
    }

    private String getText(){
        InputStream inputStream = context.getResources().openRawResource(R.raw.rules);
        BufferedReader reader = new BufferedReader(new InputStreamReader(inputStream));
        StringBuilder text = new StringBuilder();
        String line;
        try {
            while ( (line = reader.readLine()) != null) {
                text.append(line);
                text.append("\n");
            }
        }catch (IOException e){
            e.printStackTrace();
        }
        return text.toString();
    }
}

File: app/src/main/java/ir/ghararemaghzha/game/dialogs/RulesDialog.kt
package ir.ghararemaghzha.game.dialogs

import android.app.Activity
import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.Toast
import ir.ghararemaghzha.game.R
import ir.ghararemaghzha.game.classes.MySharedPreference
import ir.ghararemaghzha.game.classes.Utils
import ir.ghararemaghzha.game.data.ApiRepository
import ir.ghararemaghzha.game.data.NetworkApi
import ir.ghararemaghzha.game.data.RemoteDataSource
import ir.ghararemaghzha.game.data.Resource
import ir.ghararemaghzha.game.databinding.DialogRulesBinding
import kotlinx.coroutines.*

class RulesDialog(ctx: Context) : Dialog(ctx) {

    private lateinit var b:DialogRulesBinding
    private lateinit var job: Job

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        b = DialogRulesBinding.inflate(LayoutInflater.from(context))
        setContentView(b.root)
        job = CoroutineScope(Dispatchers.IO).launch {
            getData()
        }
        b.rulesClose.setOnClickListener { dismiss() }
    }

    private suspend fun getData() {
        val number = MySharedPreference.getInstance(context).getNumber()
        val token = MySharedPreference.getInstance(context).getAccessToken()
        if (number.isEmpty() || token.isEmpty()) {
            Utils.logout(context as Activity, true)
            return
        }
        when (val res = ApiRepository(RemoteDataSource().getApi(NetworkApi::class.java)).info("Bearer $token", number, "rules")) {
            is Resource.Success -> {
                withContext(Dispatchers.Main) {
                    b.rulesText.text = res.value.data
                }
            }
            is Resource.Failure -> {
                withContext(Dispatchers.Main) {
                    Toast.makeText(context, R.string.general_error, Toast.LENGTH_SHORT).show()
                    dismiss()
                }
            }
        }
    }

    override fun dismiss() {
        super.dismiss()
        job.cancel()
    }
}



